pkg/asset/installconfig/aws/basedomain.go

```diff
@@ -13,65 +14,85 @@ import (
 	survey "gopkg.in/AlecAivazis/survey.v1"
 )
 
+// Zone represents an AWS route53 DNS Zone
+type Zone struct {
+	ID   string
+	Name string
+}
+
 // IsForbidden returns true if and only if the input error is an HTTP
 // 403 error from the AWS API.
 func IsForbidden(err error) bool {
 	requestError, ok := err.(awserr.RequestFailure)
 	return ok && requestError.StatusCode() == http.StatusForbidden
 }
 
-// GetBaseDomain returns a base domain chosen from among the account's
-// public routes.
-func GetBaseDomain() (string, error) {
+// GetBaseDomain returns a base domain zone chosen from among the
+// account's public routes. The zone struct contains the zone ID
+// for the chosen base domain, to distinguish between multiple
+// public zones that share the same base domain.
+func GetBaseDomain() (zone *Zone, err error) {
 	session, err := GetSession()
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	logrus.Debugf("listing AWS hosted zones")
+	var publicZoneNameChoice string
```

Comment:
move this closer to the use site. like https://github.com/openshift/installer/pull/2764/commits/aefaac91c863569e84b60a856afdd895596d9677#diff-d8bb3678152879e136aec3bd82d820b2L58

Reply:
Fixed.